test/src/test/java/hudson/slaves/SlaveComputerTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018, suren <zxjlwt@126.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.slaves;
+
+import com.gargoylesoftware.htmlunit.WebResponse;
+import hudson.model.Computer;
+import hudson.model.Node;
+import hudson.security.ACL;
+import hudson.security.ACLContext;
+import hudson.security.AuthorizationStrategy;
+import hudson.security.Permission;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import org.acegisecurity.Authentication;
+import org.acegisecurity.context.SecurityContextHolder;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.xml.sax.SAXException;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * @author suren
+ */
+public class SlaveComputerTest implements Serializable {
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Test
+    public void testGetAbsoluteRemotePath() throws Exception {
+        Node nodeA = j.createOnlineSlave();
+        String path = ((DumbSlave) nodeA).getComputer().getAbsoluteRemotePath();
+        Assert.assertNotNull(path);
+        Assert.assertEquals(getRemoteFS(nodeA), path);
+
+        try(ACLContext context = ACL.as(Jenkins.ANONYMOUS)) {
+            setAsAnonymous();
+            nodeA = j.createOnlineSlave();
+            path = ((DumbSlave) nodeA).getComputer().getAbsoluteRemotePath();
+            Assert.assertNull(path);
+            Assert.assertEquals(getRemoteFS(nodeA), "null");
```

Comment:
Is `null` (the string) a reasonable output to have in the remote API? Or is that just a side effect of how `#getRemoteFS(Node)` works?

Reply:
`null` is returned by the remote API. That's the reason why I do this.

Reply:
Then you did something wrong. IIUC `getRemoteFS` in this case should be throwing `JSONException` from `getString`.

Reply:
It will not throw JSONExcpetion that under my test.